cpp/ql/lib/semmle/code/cpp/controlflow/IRGuards.qll

```diff
@@ -20,6 +20,44 @@ private predicate isUnreachedBlock(IRBlock block) {
   block.getFirstInstruction() instanceof UnreachedInstruction
 }
 
+private newtype TAbstractValue =
+  TBooleanValue(boolean b) { b = true or b = false } or
```

Comment:
Just out of curiosity: why is `b = true or b = false` needed?

Reply:
The QL type system considers booleans to be "infinite types", so `TBooleanValue(boolean b) { any() }` would be a type error since `b` appears unbounded to the type system.